server/database.js

```diff
@@ -32,16 +34,52 @@ function dataEndpoint(pool, request, response) {
     });
 }
 
-// Spoofing the part where I confirm a token is allowed to view data from 
-// a specific location. Location is currently hard coded so this is fine.
+// It seems more natural to store email/url pairs in access db.
+// Should I be using token to find the corresponding email and 
+// use the email to check for access?
+// I'm not sure how to store token/url pairs.
```

Comment:
@keving17 It looks like you figured this out and got this working, right?

Checking my understanding: the UI passes `location`, the server looks up to see if the token is valid, then looks up to see if that email is valid for that location, then sends back data for that location.  Is that right?

Reply:
yep that's right! since the UI only send the token, we have to do an extra query to look up what email corresponds to the given token but then we can use that email to see if it is authorized to grab data from that location.

Reply:
👍